src/main/java/com/maxmind/db/Reader.java

```diff
@@ -177,7 +177,12 @@ public <T> DatabaseRecord<T> getRecord(InetAddress ipAddress, Class<T> cls)
         T dataRecord = null;
         if (record > nodeCount) {
             // record is a data pointer
-            dataRecord = this.resolveDataPointer(buffer, record, cls);
+            try {
+                dataRecord = this.resolveDataPointer(buffer, record, cls);
+            } catch (DeserializationException exception) {
+                throw new DeserializationException("Error getting record for IP " + ipAddress.toString(), exception);
```

Comment:
I wonder if it would be worth including the message from the cause exception too. The primary reason is that I suspect people will create GitHub issues and not include the inner exception output.